app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision_workflow.rb

```diff
@@ -50,6 +54,38 @@ def allowed_cloud_volumes(_options = {})
     Hash[cloud_volumes || {}]
   end
 
+  def set_request_values(values)
+    values[:new_volumes] = parse_new_volumes_fields(values)
+    super
+  end
+
+  def parse_new_volumes_fields(values)
+    stop = false
+    new_volumes = []
+
+    while !stop
```

Comment:
Feel like we can do this a little better, something like
```
new_volumes = []
values.select { |k, _v| k =~ /(name|size|diskType|shareable)_(\d+)/ }.each do |key, value|
  field, cnt = key.split("_")
  cnt = Integer(cnt)

  new_volumes[cnt] ||= {}
  new_volumes[cnt][field] = value
end
```

Reply:
You could even do `/(#{volume_dialog_keys.join("|")})_(\d+)/`

Reply:
Hi. The "values" hash doesn't always contain sequential number after the '_' symbol. I.e. the stucture can look like this: 
{name_**1** => valueA, name_**3** => valueB, ...}.

Reply:
So I'm afraid we'd have to use a counter here.

Reply:
or maybe we could just add to the end of the 'new_volumes' array

Reply:
Or we could leave it like this and just do "new_volumes.compact!" after the loop. What do you think? If I'm not wrong in Ruby the unset array values between indices "a" and "b" will be "nil".

Reply:
I've tested it. "shareable" has to be explicitely checked to be not "nil" because only in that case the value must be set to "true". Also there might be an inconsistency if you add multiple volumes in one of which you'd leave all fields blank. Then UI would show all volumes up to the "blank one" but the back-end would consider all new volumes including the ones after the "blank one".